File: admin/src/main/java/com/imstargg/admin/domain/Gadget.java
package com.imstargg.admin.domain;

import com.imstargg.storage.db.core.MessageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BrawlStarsImageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.GadgetCollectionEntity;
import jakarta.annotation.Nullable;

import java.util.List;

public record Gadget(
        GadgetCollectionEntity entity,
        List<MessageCollectionEntity> names,
        @Nullable BrawlStarsImageCollectionEntity image
) {
}

File: admin/imstargg-admin-frontend/src/model/Gadget.ts
import {GadgetEntity} from "@/model/entity/GadgetEntity";
import {MessageEntity} from "@/model/entity/MessageEntity";
import { BrawlStarsImageEntity } from "./entity/BrawlStarsImageEntity";

export interface Gadget {
  entity: GadgetEntity;
  names: MessageEntity[];
  image: BrawlStarsImageEntity | null;
}